### DISCUSSION TITLE
Multiple sources + Dedicated IMU

### DISCUSSION THREAD

**Discussion by 44616E** [OpenSeizureDetector/Android_Pebble_SD#114] [04_Devices]
I'm new on epilepsy journey, with only a recent diagnosis for my child. 

I'm wondering if there would be any desire to use two sources instead of only a Garmin watch. I have been using Wellue Ring sensors for Pulse and spO2 at the same time as the Garmin watch, because I find the stability and accuracy is better than the watch. The Wellue sensors are easy to read the BLE data stream (there are projects on GitHub).

I am also wondering if we could use an IMU sensor instead of the watch, IMU would allow for better sampling rates and anti-aliasing filters and possibly more processing on the IMU itself. There are some options that could be significantly cheaper than the Garmin watch as well. 

I'm keen to do some development work if there is interest.

**Comment by jones139** [comment]
I hadn't heard of the Wellue Ring sensor, but we could fairly easily make OpenSeizureDetector work with it if there are open source examples of reading the data from the device - making a companion app that sends data to the OpenSeizueDetector android app via http would be the easiest way of doing it.   The only change we'd make to the android app is to have the option of ignoring the Garmin Heart Rate and SPO2 data if we are using a different sensor device for that.

**Reply by 44616E** [reply]
It appears we should also be able to read the ring from the watch. But I would prefer to use BLE to the phone for reliability and watch battery life concerns.

**Comment by jones139** [comment]
It is definitely possible to make another custom device to send data to the phone instead of a Garmin watch - I got as far as a proof of concept here:  https://github.com/OpenSeizureDetector/OSDWatch and https://www.openseizuredetector.org.uk/?p=1115

The issue is it will be a lot of effort, especially to get it small enough to fit in a watch form factor and have acceptable battery life, so from my point of view, it would have to do something that we can not do with a commercial device.    I stopped working on it when I discovered the PineTime watch, which had most of the sensor devices on it that I would have put on my own watch - see https://www.openseizuredetector.org.uk/?page_id=1480#PineTime and https://github.com/orgs/OpenSeizureDetector/discussions/70

**Comment by jones139** [comment]
It might well be that using the external HR and O2SAT sensors will improve
battery life because the LEDs in the watch sensor will not need to light.
 There is a secondary advantage of using the watch as the interface which
is that there will be no phone software changes required.

On Thu, 27 Jul 2023 at 12:11, 44616E ***@***.***> wrote:

> It appears we should also be able to read the ring from the watch. But I
> would prefer to use BLE to the phone for reliability and watch battery life
> concerns.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OpenSeizureDetector/discussions/114#discussioncomment-6564137>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACLSYZRSHURJOJQQIS2KR3XSJEGJANCNFSM6AAAAAA2TV24OU>
> .
> You are receiving this because you commented.Message ID:
> <OpenSeizureDetector/Android_Pebble_SD/repo-discussions/114/comments/6564137
> @github.com>
>


-- 
Graham Jones <http://google.com/+GrahamJones>
Hartlepool, UK.

**Comment by 44616E** [comment]
Will need software changes in the watch app, as they are not supported sensors. The watch also won't stay connected to sensors by default. Its an option.
